pandas/core/internals/managers.py

```diff
@@ -351,8 +351,12 @@ def where(self: T, other, cond, align: bool, errors: str) -> T:
             errors=errors,
         )
 
-    def setitem(self: T, indexer, value) -> T:
-        return self.apply("setitem", indexer=indexer, value=value)
+    def setitem(self: T, indexer, value, inplace=False) -> T:
```

Comment:
why is this change necessary?

Reply:
To follow the change that was needed in the ArrayManager (to ensure every mutation of the values happens in the manager), see https://github.com/pandas-dev/pandas/pull/41879 for this change as a separated pre-cursor PR